cdap-api/src/main/java/co/cask/cdap/api/data/stream/StreamBatchReadable.java

```diff
@@ -1,339 +0,0 @@
-/*
- * Copyright © 2014-2016 Cask Data, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License"); you may not
- * use this file except in compliance with the License. You may obtain a copy of
- * the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations under
- * the License.
- */
-
-package co.cask.cdap.api.data.stream;
-
-import co.cask.cdap.api.annotation.Beta;
-import co.cask.cdap.api.data.batch.BatchReadable;
-import co.cask.cdap.api.data.batch.Input;
-import co.cask.cdap.api.data.batch.Split;
-import co.cask.cdap.api.data.batch.SplitReader;
-import co.cask.cdap.api.data.format.FormatSpecification;
-import co.cask.cdap.api.data.schema.Schema;
-import co.cask.cdap.api.flow.flowlet.StreamEvent;
-import co.cask.cdap.api.mapreduce.MapReduceContext;
-import co.cask.cdap.api.stream.GenericStreamEventData;
-import co.cask.cdap.api.stream.StreamEventDecoder;
-import co.cask.cdap.internal.io.SchemaTypeAdapter;
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
-
-import java.io.UnsupportedEncodingException;
-import java.net.URI;
-import java.net.URLDecoder;
-import java.net.URLEncoder;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import javax.annotation.Nullable;
-
-/**
- * This class is for using a Stream as input for a MapReduce program. 
- * 
- * <p>
- * An instance of this class should be set in the
- * {@link MapReduceContext} of the {@code beforeSubmit} method to use the Stream as input:
- * </p>
- * <pre><code>
- *   class MyMapReduce implements MapReduce {
- *       public void beforeSubmit(MapReduceContext context) {
- *         context.addInput(Input.ofStream("mystream"));
- *       }
- *   }
- * </code></pre>
- *
- * @deprecated as of version 3.4.0. Use {@link MapReduceContext#addInput(Input)}
- */
-@Deprecated
-public class StreamBatchReadable implements BatchReadable<Long, String> {
```

Comment:
most of it centers around converting the required information into a URI, which was only needed back when you could set a stream as input in the configure method of a mapreduce. Today its only possible at initialize time so none of it is needed anymore.